packages/core/src/test/interpolate.test.ts

```diff
@@ -15,42 +17,63 @@ test('Must be the same length', () => {
 	}, /inputRange \(2\) and outputRange \(3\) must have the same length/);
 });
 
-test('Test against Infinity values', () => {
-	expectToThrow(() => {
-		interpolate(1, [-Infinity, Infinity], [0, 2]);
-	}, /inputRange must contain only finite numbers, but got \[-Infinity,Infinity\]/);
-});
-
-test('Must pass at least 2 elements', () => {
+test('Must pass at least 2 elements for input range', () => {
 	expectToThrow(() => {
 		interpolate(1, [0], [9]);
 	}, /inputRange must have at least 2 elements/);
 });
 
-test('Input range must be monotonically non-decreasing', () => {
+test('Input range must be strictly monotonically non-decreasing', () => {
 	expectToThrow(() => {
 		interpolate(1, [0, 1, 0.5], [0, 2, 0.2]);
-	}, /inputRange must be monotonically non-decreasing/);
+	}, /inputRange must be strictly monotonically non-decreasing/);
+	expectToThrow(() => {
+		interpolate(0.75, [0, 1, 1], [0, 2, 0]);
+	}, /inputRange must be strictly monotonically non-decreasing/);
 });
 
-test('Output range can be monotonically decreasing', () => {
+test('Output range strictly monotonically decreasing', () => {
 	expect(interpolate(0.75, [0, 0.5, 1], [0, 2, 0])).toEqual(1);
```

Comment:
The test name is (and was) kind of bad here, what we are testing here is that output range doesn't need to have any monotonic increase/decrease properties at all, it works if it is increasing then decreasing again.